core/src/main/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbe.java

```diff
@@ -63,59 +70,77 @@ public class HasUnreleasedProductionChangesProbe  extends Probe {
-            final LogCommand logCommand = git.log();
+            final LogCommand logCommand = git.log().setMaxCount(1);
             if (folder != null) {
                 logCommand.addPath(folder);
             }
-
-            for (String path : productionPathsToCheckForCommits) { logCommand.addPath(path); }
-
+            productionPathsToCheckForCommits.forEach(logCommand :: addPath);
             Iterable<RevCommit> commits = logCommand.call();
 
-            if (commits == null) {
-                return ProbeResult.success(key(), "All production modifications were released.");
-            }
-
             for( RevCommit commit : commits ) {
                 Instant instant = Instant.ofEpochSecond(commit.getCommitTime());
                 ZonedDateTime zonedDateTime = ZonedDateTime.ofInstant(instant, ZoneId.systemDefault());
 
-                System.out.println(commit.getName() + " "+zonedDateTime);
-
-
-//                final ZonedDateTime commitDate = ZonedDateTime.ofInstant(
-//                    commit.getAuthorIdent().getWhenAsInstant(),
-//                    commit.getAuthorIdent().getZoneId()
-//                );
-                if (zonedDateTime.isAfter(plugin.getReleaseTimestamp())) {
-                    System.out.println("release timestamp= "+plugin.getReleaseTimestamp());
-                    final TreeWalk walk = new TreeWalk(git.getRepository());
-                    walk.setRecursive(true);
-                    walk.addTree(commit.getTree());
-                    while (walk.next()) {
-                        commitFileList.add(walk.getPathString());
-                        System.out.println(walk.getPathString() + " "+zonedDateTime);
+                if(commit.getParentCount() > 0 ) {
+                    /*
+                    *  if a previous commit exists, compare the difference
+                    * */
+
+                    // https://stackoverflow.com/a/27375013/9493145
+
+                    ObjectReader reader = git.getRepository().newObjectReader();
+                    CanonicalTreeParser oldTreeIter = new CanonicalTreeParser();
+                    ObjectId oldTree = git.getRepository().resolve( "HEAD^{tree}" ); // equals newCommit.getTree()
+                    oldTreeIter.reset( reader, oldTree );
+                    CanonicalTreeParser newTreeIter = new CanonicalTreeParser();
+                    ObjectId newTree = git.getRepository().resolve( "HEAD~1^{tree}" ); // equals oldCommit.getTree()
+                    newTreeIter.reset( reader, newTree );
+
+                    DiffFormatter df = new DiffFormatter( new ByteArrayOutputStream() ); // use NullOutputStream.INSTANCE if you don't need the diff output
+                    df.setRepository( git.getRepository() );
+                    List<DiffEntry> entries = df.scan( oldTreeIter, newTreeIter );
+
+                    for( DiffEntry entry : entries ) {
+                        System.out.println( entry );
```

Comment:
Hey @alecharp 

I have tried to reproduce most of the code from our last meeting. 

I read about using JGit `diff`. It requires two commits to compare.

In our case, we need to find the commit after the latest release. 

How do I compare time here?

Reply:
You don't. 
You need to iterate over the commits of the repository and stop as soon as you find a commit which was made before the release. 

For each commit, you need to see if it has a parent. If so, you can use that parent to do a diff. otherwise you can simply browse the repository content with a `TreeWalk`